### ISSUE TITLE
Fuzz subprocesses can import another checkout and misclassify dependency frames

### ISSUE THREAD

**Issue by glenn-jocher** [ultralytics/ultralytics#25293]
### Search before asking

- [x] I have searched the Ultralytics issues and found no similar bug report.

### Ultralytics version

Latest `main`

### Environment

Editable installs with multiple Git worktrees; Linux and Windows fuzz runners

### Bug

`.github/scripts/fuzz.py` launches each CLI trial from a temporary working directory while inheriting the ambient Python import path. With another editable checkout installed, a repro launched from a feature worktree can therefore execute code from the primary checkout.

Traceback parsing also treats dependency paths below a checkout named `ultralytics` as package frames. For example, both `ultralytics/.venv/lib/.../site-packages/torch/...` and the Windows `ultralytics/.venv/Lib/.../site-packages/torch/...` enter the signature, producing platform-dependent titles and duplicate findings for the same Ultralytics failure.

### Expected behavior

Fuzz trials should import the checkout containing the invoked script, and signatures should contain Ultralytics package frames rather than virtual-environment dependency frames.

### Minimal reproduction

Run `.github/scripts/fuzz.py` from a linked worktree while the primary checkout is installed editable, then inspect a child CLI traceback. Parse equivalent Linux and Windows tracebacks containing a Torch frame below `<checkout>/.venv/.../site-packages/`; the generated signatures differ.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for reporting this issue and for providing a clear reproduction involving editable installs, Git worktrees, and platform-dependent traceback parsing 🚀

This is an automated response. An Ultralytics engineer will also assist soon. Your reported minimal reproduction is helpful; please include the exact commands, relevant environment details, and a complete child CLI traceback or patch if available so we can reproduce the behavior consistently on Linux and Windows.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for new users, including [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples, where many common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) if anything beyond the details already included is needed.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by UltralyticsAssistant** [comment]
A potential fix has been merged in [PR #25294](https://github.com/ultralytics/ultralytics/pull/25294) by [@glenn-jocher](https://github.com/glenn-jocher), titled **“Fix fuzz subprocess source and signatures.”**

The changes improve fuzz testing by:

- Adding the repository root to `PYTHONPATH` so fuzz trials run against the local Ultralytics source rather than an installed package.
- Normalizing Windows and Unix-style paths in traceback parsing.
- Including only traceback frames from the local `ultralytics` source.
- Excluding unrelated `site-packages` frames.
- Producing consistent `ultralytics/...` frame paths for clearer crash reports.

These updates should make fuzz testing more accurate and help identify reproducible issues within the code currently under development. Credit to [@glenn-jocher](https://github.com/glenn-jocher) and the contributors involved in the PR.

To test the latest changes directly from the repository:

```bash
pip install git+https://github.com/ultralytics/ultralytics.git@main
```

Alternatively, you can wait for the next official PyPI release.

Please let us know whether these changes resolve the issue. Thank you 🙏 for reporting it, and any further feedback is welcome if the problem persists.
